codelab-master/apps/playground/src/app/quiz/components/question/question.component.html

```diff
@@ -0,0 +1,82 @@
+<ng-content></ng-content>
+
+<form [formGroup]="formGroup">
+  <div class="options" *ngFor="let option of currentQuestion.options; index as i">
+    <div class="single-answer" *ngIf="!multipleAnswer">
+      <mat-radio-button
+        (change)="setSelected(i)"
+        [disabled]="alreadyAnswered"
+        [class.is-correct]="option.selected && option.correct"
+        [class.is-incorrect]="option.selected && !option.correct">
+
+        <mat-radio-label>
+          <span>{{ i + 1 }}. {{ option.text }}</span>
+
+          <mat-icon class="feedback-icon"
+                    *ngIf="option.selected && option.correct && isCorrectAnswerSelected">done</mat-icon>
+          <mat-icon class="feedback-icon"
+                    *ngIf="option.selected && !option.correct">clear</mat-icon>
+        </mat-radio-label>
+      </mat-radio-button>
+
+      <section class="messages">
+        <div *ngIf="option.selected && option.correct && isCorrectAnswerSelected">
+          <span>
+            <mat-icon class="sentiment-icon">sentiment_very_satisfied</mat-icon>&nbsp;&nbsp;&nbsp;
+            <span class="message correct-message" i18n>&nbsp;&nbsp;&nbsp;You&apos;re right! {{ correctMessage }}</span>
+          </span>
+        </div>
+        <div *ngIf="option.selected && !option.correct">
+          <span>
+            <mat-icon class="sentiment-icon">sentiment_very_dissatisfied</mat-icon>&nbsp;&nbsp;&nbsp;
+            <span class="message wrong-message" i18n>&nbsp;&nbsp;&nbsp;That&apos;s wrong. {{ correctMessage }}</span>
+          </span>
+        </div>
+      </section>
+      <!-- TODO: get this working, except with previousUserAnswersText -->
+      <!-- <div *ngFor="let previousOption of previousUserAnswers[i]" class="previous-answer-text">
+        <span *ngIf="option.selected && previousUserAnswers[i].length === 1">
+          Option {{ previousOption }} was selected in your previous quiz.
+        </span>
+      </div> -->
+    </div>
+
+    <div class="multiple-answer" *ngIf="multipleAnswer">
```

Comment:
Should there be two different components for single and multiple answer?

Reply:
I'm trying to separate them into individual components for single-answer and multiple-answer. Will probably need to pass the variables like multipleAnswer and option from the QuizQuestionComponent to the new components.

Reply:
I'm trying to fix the CSS on the mat-checkbox form yet again. The width of the options/mat-checkboxes doesn't seem to want to display at 630px, but oddly the mat-radio-button form seems to be displaying fine.

Reply:
@NothingEverHappens, should there be a QuestionService specifically to share the functions and properties for the MultipleAnswerComponent and SingleAnswerComponent? Please let me know.

Reply:
ok navigation works (sort of), am refactoring the code to use observables....

Reply:
Okay navigation works, though now I am getting red highlighting on the options of the multiple-answer question when the questions are shuffled.

Reply:
This is because I have renamed the ng-if with a variable that holds the text of the ng-if in the service, as you mentioned in a different issue. If I just have:
 <mat-checkbox
      (change)="setSelected(i)"
      [class.is-correct]="option.selected && option.correct"
      [class.is-incorrect]="option.selected && !option.correct">
then I don't get the red highlighting when shuffling but the text is somehow getting shifted towards the center of the box

Reply:
In the single-answer questions, if I replace the ngClass with the isCorrectOption and isIncorrectOption, it highlights all red, yet in the multiple-answer questions, if I replace with isCorrectOption/isIncorrectOption it doesn't seem to highlight at all which is what I want for the single-answer. I tried in the resetAll to reset the isCorrectOption/isIncorrectOption to empty strings, but that didn't work.

Reply:
OK now having an issue with correctOptions - it doesn't show the correct option(s) for next question and also in ResultsComponent the correct options aren't displaying correctly.

Reply:
ok added correctMessage to ngOnChanges, will look at ResultsComponent soon...

Reply:
I reduced the code in the question components by using an abstract class (QuizQuestionComponent) and am running into 3 circular dependencies warnings.